pkg/controller/utils/utils.go

```diff
@@ -352,14 +352,27 @@ func GetAuthentication(ctx context.Context, cli client.Client) (*operatorv1.Auth
 	return authentication, nil
 }
 
-// GetInstallationStatus returns the current installation status, for use by other controllers.
-func GetInstallationStatus(ctx context.Context, client client.Client) (*operatorv1.InstallationStatus, error) {
+// GetInstallationAndStatus returns the current installation status, for use by other controllers.
+func GetInstallationAndStatus(ctx context.Context, client client.Client) (*operatorv1.InstallationSpec, *operatorv1.InstallationStatus, error) {
 	// Fetch the Installation instance. We only support a single instance named "default".
 	instance := &operatorv1.Installation{}
 	if err := client.Get(ctx, DefaultInstanceKey, instance); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	return &instance.Status, nil
+
+	spec := instance.Spec
+
+	// update Installation with 'overlay'
+	overlay := operatorv1.Installation{}
```

Comment:
I don't like the duplication of this code with GetInstallation. The comment doesn't mention that it accounts for overlays.
Is the rationale that it is a good trade off to duplicate this code to avoid a client.Get for the same resource?

Reply:
Yes. The rationale was to avoid another client.Get. @rene-dekker

Reply:
It was my suggestion to return both the status and the installation object. I think it is costly to hold up the PR over this though, so if you think it is a bad idea, I am ok to either:
- make two calls: GetInstallation + a call for the status.
- have the GetInstallation reuse this func to reduce the overlay code duplication.
- add an extra returned object to getInstallation(). (Maybe that would be best done in a follow up PR)

Reply:
I think it is a great idea to have GetInstallation call this function. I can totally be on board with that. Plus a comment update that this function handles the the overlay also.

Reply:
client.Get() should be pretty darn cheap - it uses the cache first before hitting the API server so I would expect most cases there's not even an API call

Reply:
I have a used a separate function that just returns the status.